solr/core/src/java/org/apache/solr/core/PluginBag.java

```diff
@@ -288,6 +294,20 @@ public PluginHolder<T> put(String name, PluginHolder<T> plugin) {
     return old;
   }
 
+  static final class APIConfigProviderBinder extends AbstractBinder {
+
+    private final APIConfigProvider<?> cfgProvider;
+
+    public APIConfigProviderBinder(APIConfigProvider<?> cfgProvider) {
+      this.cfgProvider = cfgProvider;
+    }
+
+    @Override
+    protected void configure() {
+      bindFactory(cfgProvider).to(cfgProvider.getConfigClass());
```

Comment:
I agree with the perf aspect, this will get called on each call. my thought was that the config object gets created once and simply returned as many times as needed. I would suggest we leave as is and revisit this pattern if needed, but let me double check again on the scope parts.